### DISCUSSION TITLE
After switching from default to K8s run launcher, why is my job stuck in the starting state?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17619] [Q&A]
Hi everyone,
Changed my run_launcher from DefaultRunLauncher to K8sRunLauncher and facing the issue of job being stuck in starting state. Screenshot is attached for refrence. Please suggest what could be the reason and how can i resolve the issue.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1698866572229889?thread_ts=1698866572.229889&cid=C01U954MEER).*

**Comment by shalabhc** [comment]
If you Kubernetes job is stuck in starting state, the best way to debug it is to look at the kubernetes logs. I see in the slack screenshot that it prints the Kubernetes Job name. You want to use `kubectl` or another tool to print the logs and events for that job. That should show why the job is stuck or not making progress.
